### PR TITLE
feat: Support running diffs against prior commits of the same branch

### PR DESCRIPTION
**Pull request by armenzg**
This is in order to support running Visual Snapshots on the commits on Sentry's `master` branch rather than on PRs in order to reduce load.

This project will also require running VS on-demand for PRs but it's separate work from this PR.

### CODE REVIEW

**Comment by armenzg** [comment]
This shows a PR run that did not regress with this change:
https://github.com/getsentry/sentry/runs/6835937666?check_suite_focus=true#step:3:80

> Run getsentry/action-visual-snapshot@run-on-branches

**Comment by armenzg** [comment]
I have been able to get visual diffs for per-commit runs on [this branch](https://github.com/getsentry/sentry/commits/execute-vs-on-a-branch).

I forced a VS regression and I managed to approve it.
See [url](https://github.com/getsentry/sentry/runs/6820297206).

You can see that it picked up the `execute-vs-on-a-branch` branch rather than `master`.
> getArtifactsForBranchAndWorkflow - workflow:"acceptance.yml",  branch:"execute-vs-on-a-branch"

https://github.com/getsentry/sentry/runs/6820293972?check_suite_focus=true#step:3:308

**Comment by billyvg** [comment]
I would say if we are going to turn this off for PRs then we should remove it completely. It won't have much purpose running only on master

**Comment by armenzg** [comment]
The idea is that we will still be able to run it on PRs on request.

The execution on `master` would help us know where a regression is introduced.

File: src/api/getArtifactsForBranchAndWorkflow.ts [hunk 1]
```diff
@@ -78,7 +78,7 @@ export async function getArtifactsForBranchAndWorkflow(
     //
     // If this needs to be more generic, this should be an option.
     const workflowRuns = response.data.filter(
-      workflowRun =>
+      (workflowRun: any) =>
```
[the remark this answers is not in the document]
Comment (billyvg) [reply src/api/getArtifactsForBranchAndWorkflow.ts:81]: Was it an error or just a warning? It should be able to infer the type

**Review by billyvg — APPROVED** [review]

File: src/api/getArtifactsForBranchAndWorkflow.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (armenzg) [reply src/api/getArtifactsForBranchAndWorkflow.ts]: Maybe I had old packages installed. I reverted it and I don't see the issue or warning anymore.

**Comment by armenzg** [comment]
Regardless if we end up changing the scheduling approach, this help me deal with changes that would require me to have another branch that pretends to be `master`.

**Pull request merged**